### ISSUE TITLE
[BUG] Drawer failing in controlled operations

### ISSUE THREAD

**Issue by KetpuntoG** [PennyLaneAI/pennylane#5064]
### Expected behavior

I would expect the following code to draw the circuit correctly.

```python
def U(wires):
    qml.adjoint(qml.CNOT)(wires = wires)

dev = qml.device("default.qubit")

@qml.qnode(dev)
def circuit():
  
    qml.ctrl(U, control = 0)(wires = [1,2])

    return qml.state()

qml.draw_mpl(circuit)()
```

### Actual behavior

The code provided does not work correctly.

```
`ValueError: `control_values` must be the same length as `wires``
```

### Additional information

It does work with the text drawer

### System information

```shell
Name: PennyLane
Version: 0.34.0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/PennyLaneAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /usr/local/lib/python3.10/dist-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, requests, rustworkx, scipy, semantic-version, toml, typing-extensions
Required-by: PennyLane-Lightning

Platform info:           Linux-6.1.58+-x86_64-with-glibc2.35
Python version:          3.10.12
Numpy version:           1.23.5
Scipy version:           1.11.4
Installed devices:
- lightning.qubit (PennyLane-Lightning-0.34.0)
- default.gaussian (PennyLane-0.34.0)
- default.mixed (PennyLane-0.34.0)
- default.qubit (PennyLane-0.34.0)
- default.qubit.autograd (PennyLane-0.34.0)
- default.qubit.jax (PennyLane-0.34.0)
- default.qubit.legacy (PennyLane-0.34.0)
- default.qubit.tf (PennyLane-0.34.0)
- default.qubit.torch (PennyLane-0.34.0)
- default.qutrit (PennyLane-0.34.0)
- null.qubit (PennyLane-0.34.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by astralcai** [comment]
This seems to be caused by the `Adjoint` of `CNOT` also having control wires, and in `unwrap_controls` method of `tape_mpl.py`, the control wire of `Adjoint` and the control wire of `CNOT` are double-counted.

```
>>> op = qml.adjoint(qml.CNOT)(wires = [0,1])
>>> op.control_wires
<Wires = [0]>
```

I'm looking into why it's doing this. A simple solution would be to check for duplicated wire labels in control_wires. I can work on a fix if nobody else is already working on this.

**Comment by astralcai** [comment]
Essentially the issue is caused by operators having a `control_wire` property despite not actually being controlled operators. I am working on a separate PR that make all controlled operators actually inherit from `Controlled`. Once that is complete, a simple `is_instance` check would fix the issue.
